operator/internal/manifests/config_otlp.go

```diff
@@ -0,0 +1,325 @@
+	if len(cfg.DefaultIndexLabels) > 1 {
+		slices.Sort(cfg.DefaultIndexLabels)
+		cfg.DefaultIndexLabels = slices.Compact(cfg.DefaultIndexLabels)
+	}
+
+	if cfg.Global != nil {
+		if len(cfg.Global.ResourceAttributes) > 1 {
+			cfg.Global.ResourceAttributes = sortAndDeduplicateOTLPAttributes(cfg.Global.ResourceAttributes)
+		}
+
+		if len(cfg.Global.ScopeAttributes) > 1 {
+			cfg.Global.ScopeAttributes = sortAndDeduplicateOTLPAttributes(cfg.Global.ScopeAttributes)
+		}
+
+		if len(cfg.Global.LogAttributes) > 1 {
+			cfg.Global.LogAttributes = sortAndDeduplicateOTLPAttributes(cfg.Global.LogAttributes)
+		}
+	}
+
+	for _, t := range cfg.Tenants {
+		if len(t.ResourceAttributes) > 1 {
+			t.ResourceAttributes = sortAndDeduplicateOTLPAttributes(t.ResourceAttributes)
+		}
+
+		if len(t.ScopeAttributes) > 1 {
+			t.ScopeAttributes = sortAndDeduplicateOTLPAttributes(t.ScopeAttributes)
+		}
+
+		if len(t.LogAttributes) > 1 {
+			t.LogAttributes = sortAndDeduplicateOTLPAttributes(t.LogAttributes)
+		}
+	}
+
+	return cfg
+}
+
+func sortAndDeduplicateOTLPAttributes(attrs []config.OTLPAttribute) []config.OTLPAttribute {
+	slices.SortFunc(attrs, func(a, b config.OTLPAttribute) int {
+		action := strings.Compare(string(a.Action), string(b.Action))
+		if action != 0 {
+			return action
+		}
+
+		if a.Regex != "" && b.Regex != "" {
+			return strings.Compare(a.Regex, b.Regex)
+		}
+
+		if a.Regex != "" && b.Regex == "" {
+			return 1
+		}
+
+		if a.Regex == "" && b.Regex != "" {
+			return -1
+		}
+
+		return 0
+	})
+
+	for i := 0; i < len(attrs)-1; i++ {
+		a := attrs[i]
+		if a.Regex != "" {
+			continue
+		}
+
+		slices.Sort(a.Names)
+		attrs[i] = a
+
+		next := attrs[i+1]
+		if next.Regex != "" {
+			continue
+		}
+
+		if a.Action != next.Action {
+			continue
+		}
+
+		// Combine attribute definitions if they have the same action and just contain names
+		a.Names = append(a.Names, next.Names...)
+		slices.Sort(a.Names)
+		a.Names = slices.Compact(a.Names)
+
+		// Remove the "next" attribute definition
+		attrs[i] = a
+		attrs = append(attrs[:i+1], attrs[i+2:]...)
+		i--
+	}
+
+	return attrs
+}
+
+func otlpAttributeConfig(ls *lokiv1.LokiStackSpec) config.OTLPAttributeConfig {
+	result := defaultOTLPAttributeConfig(ls.Tenants)
+
+	if ls.Limits != nil {
+		if ls.Limits.Global != nil && ls.Limits.Global.OTLP != nil {
+			globalOTLP := ls.Limits.Global.OTLP
+
+			if globalOTLP.StreamLabels != nil {
+				regularExpressions := []string{}
+				for _, attr := range globalOTLP.StreamLabels.ResourceAttributes {
+					if attr.Regex {
+						regularExpressions = append(regularExpressions, attr.Name)
+						continue
+					}
+
+					result.DefaultIndexLabels = append(result.DefaultIndexLabels, attr.Name)
+				}
+
+				if len(regularExpressions) > 0 {
+					result.Global = &config.OTLPTenantAttributeConfig{}
+
+					for _, re := range regularExpressions {
+						result.Global.ResourceAttributes = append(result.Global.ResourceAttributes, config.OTLPAttribute{
```

Comment:
Since we put regularExpressions on the `result.Global.ResourceAttributes` instead of `result.DefaultIndexLabels` (since this is only an array of strings) then when a tenant defines `IgnoreGlobalStreamLabels` these regexes will still land there no? At least from what I understand from `ignore_defaults` is that it only works to ignore the `default_resource_attributes_as_index_labels`